app/api/schema/discount_codes.py

```diff
@@ -90,6 +92,36 @@ def validate_quantity(self, data, original_data):
                 raise UnprocessableEntity({'pointer': '/data/attributes/tickets-number'},
                                           "tickets-number should be greater than max-quantity")
 
+    @validates_schema(pass_original=True)
+    def validate_value(self, data, original_data):
+        if 'id' in original_data['data']:
+            try:
+                discount_code = DiscountCode.query.filter_by(id=original_data['data']['id']).one()
+            except NoResultFound:
+                raise ObjectNotFound({'parameter': '{code}'}, "DiscountCode: not found")
+
+            if 'type' not in data:
+                data['type'] = discount_code.type
+
+            if 'value' not in data:
+                data['value'] = discount_code.value
+
+        if data['type'] == "percent":
+            if data['value'] > 100:
+                raise UnprocessableEntity(
+                    {'pointer': '/data/attributes/value'},
+                    "discount percent cannot be more than 100%"
+                )
+
+        if data['type'] == "amount":
+            event_object = Event.query.filter_by(id=data['event']).one()
+            for ticket in event_object.tickets:
```

Comment:
Can't a discount code be valid only for certain tickets?

Reply:
@iamareebjamal when the discount codes are used for `tickets` then the check in class `DiscountCodeSchemaTicket` rather than these checks which are in `DiscountCodeSchemaEvent`. The checks in `DiscountCodeSchemaEvent` are called only when discount codes will be used for events (not tickets).